### DISCUSSION TITLE
How to build a project as a new contributor

### DISCUSSION THREAD

**Discussion by GintasS** [gitextensions/gitextensions#8507] [General] [answered]
## Current behaviour
I want to contribute to gitextensions, but I can't seem to get the solution to build, is there any concrete guide available, because https://github.com/gitextensions/gitextensions/wiki/How-To:-build-instructions is not that detailed at all, I googled a few things, including issues from this repository, so found out that I need Wix, I installed that, also individual components on visual studio, still getting: 
Externals\ICSharpCode.TextEditor\Project\ICSharpCode.TextEditor.csproj'. Check that the project reference is valid and that the project file exists.	

and much more errors related to the references.

A few examples:

![image](https://user-images.githubusercontent.com/20372363/94959255-8afaeb80-04f9-11eb-9137-f147cd2e59a3.png)
![image](https://user-images.githubusercontent.com/20372363/94959271-94845380-04f9-11eb-8500-fd4bbc2e87bd.png)

I appreciate any help and sorry if this post is not meant to be here.

## Environment

- GitExtensions version:
- GIT version: 
- OS version: 
- .NET version:

**Comment by pmiossec** [comment]
I think that is because you didn't initialized the git submodules.

Once you did it, that should build without problems.

**Reply by LailaLilo** [reply]
@pmiossec  I'm getting the same issue as the OP, "you didn't initialized the git submodules.", what/where are these git submodules?
The compilation documentation says "⚠️ Make sure to synchronize and update all submodules using GitExtensions."
But I'm not sure how to synchronize/update this modules with GitExtensions, i dont even know where are them.

**Reply by gerhardol** [reply]
![image](https://github.com/user-attachments/assets/c07bf24e-bf81-427b-86d6-e07168769096)

**Comment by pmiossec** [comment]
Wix is needed only for building the setup so you don't need it to just
build and contribute...

**Reply by RussKie** [reply]
This is something that can be automated with [*.vsconfig file](https://docs.microsoft.com/en-us/visualstudio/install/import-export-installation-configurations?view=vs-2019).

**Comment by gerhardol** [comment]
Converted this to a discussion.
Added a note about WiX, submodules were already mentioned (marked with a warning too)

**Reply by RussKie** [reply]
Thank you. I further added instructions, please have a look.

**Comment by RussKie** [comment, marked as answer]
@GintasS thank you for the feedback. Please have a look at the updates to the instructions and let us know if there is anything else worth adding to make them easier to follow.

**Reply by GintasS** [reply]
Sorry for not responding that long, yes I think it's better!Thanks!
